ppx/browser/ppx_deriving_json_runtime.ml

```diff
@@ -3,8 +3,26 @@ type t = Js.Json.t
 let to_json t = t
 let of_json t = t
 let to_string t = Js.Json.stringify t
-let of_string s = Js.Json.parseExn s
-let of_json_error msg = raise @@ Json.Decode.DecodeError msg
+
+exception Json_error of string
+
+let of_string s =
+  try Js.Json.parseExn s
+  with exn ->
+    let msg =
+      match Js.Exn.asJsExn exn with
+      | Some jsexn -> Js.Exn.message jsexn
+      | None -> None
+    in
+    let msg =
+      (* msg really cannot be None in browser or any sane JS runtime *)
```

Comment:
The reason is because the exn raised by `Js.Json.parseExn` is a JavaScript one, right?

Reply:
yes